File: frontend-main/src/models/rally/locationMarker.ts
export abstract class LocationMarker {
  id: string;

  constructor(id: string) {
    this.id = id;
  }
}

export class IBeaconLocationMarker extends LocationMarker {
  uuid: string;
  major: number;
  minor: number;

  constructor(id: string, uuid: string, major: number, minor: number) {
    super(id);
    this.uuid = uuid;
    this.major = major;
    this.minor = minor;
  }
}

export class QRCodeLocationMarker extends LocationMarker {
  content: string;

  constructor(id: string, content: string) {
    super(id);
    this.content = content;
  }
}

File: flutter-application-main/lib/models/rally/location_marker.dart
import 'package:dart_mappable/dart_mappable.dart';

@MappableClass(discriminatorKey: "type")
abstract class LocationMarker {
  final String id;

  LocationMarker(this.id);
}

@MappableClass(discriminatorValue: "IBeaconLocationMarker")
class IBeaconLocationMarker extends LocationMarker {
  final String uuid;
  final int? major;
  final int? minor;

  IBeaconLocationMarker(
    String id, {
    required this.uuid,
    required this.major,
    required this.minor,
  }) : super(id);
}

@MappableClass(discriminatorValue: "QRCodeLocationMarker")
class QRCodeLocationMarker extends LocationMarker {
  final String content;

  QRCodeLocationMarker(String id, {required this.content}) : super(id);
}
